pkg/iniparser/iniparser.go

```diff
@@ -0,0 +1,161 @@
+package iniparser
+
+//importing the needed packages
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// defining the errors that may happen
+var ErrNotINI = errors.New("not an INI file")
+var ErrSectionNotThere = errors.New("section not there")
+var ErrKeyAlreadyExists = errors.New("for the provided section, the provided key already exists")
+var ErrKeyNotThere = errors.New("key not there")
+var ErrReadingFile = errors.New("can't read file")
+var ErrOpeningFile = errors.New("can't open file")
+var ErrInvalidContent = errors.New("invalid content to be stored in an ini parser")
+var ErrSectionAlreadyThere = errors.New("there is already a section with the provided name")
+var ErrWritingToFile = errors.New("problem when writing to file")
+
+// defining a data type for the ini parser
+type IniParser struct {
+	data map[string](map[string]string)
+}
+
+// initialize a new ini parser and returning a pointer to it
+func NewIniParser() *IniParser {
+	myIniParser := IniParser{make(map[string]map[string]string)}
+	return &myIniParser
+}
+
+// given a string, call the helper to check for valid ini parser content and add the content to the parser
+func (p *IniParser) LoadFromString(str string) error {
+	err := p.parse(str)
+	// fmt.Println(p.data)
+	fmt.Println(err)
+	return err
+}
+
+// given a file path , call the helper to check for valid ini parser content and add the content to the parser
+func (p *IniParser) LoadFromFile(file string) error {
+	if !strings.HasSuffix(file, ".ini") {
+		return ErrNotINI
+	}
+	data, e := os.ReadFile(file)
+	stringData := string(data)
```

Comment:
In case of error setting the data here has no use, as it will return the ErrReadingFile error, please set it after the error condition